### DISCUSSION TITLE
[v3]Custom wire:model with $dispatch

### DISCUSSION THREAD

**Discussion by jesperbjoernandersen** [livewire/livewire#6421] [7. Bugs (archived)]
### Livewire version

v3

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. Create a div with wire:model
2. Create a child div that has wire:click="$dispatch('input', 'value')"

### Additional information

In v2, you could attach wire:model to any type of element and have any child element fire off an event with $dispatch, to fake an input event, which would then trigger wire:model on the parent.

This no longer seems to be working in V3. I'm not sure if it's a bug or if it's intended, to have people use more "custom" named events like $dispatch('button-clicked', 'value') and then use the #[On('button-clicked')] Attribute instead, to call a function.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by PhiloNL** [comment]
I'm not 100% sure, but I don't think it was ever intended that you could add `wire:model` to a div, but maybe I'm wrong 🫣 

I'm curious why not use `wire:click="$set('propertyName', 'value')"`?

**Reply by jesperbjoernandersen** [reply]
Caleb made an entire screencast for v2, where he was showing that you could use $dispatch on any element and then hook into it with wire:model, also on any element. So my guess it, it was intended.

Because I'm making a reusable component, where you can just use wire:model on the component itself to wire it to anything you want. e.g. <x-component wire:model="whatever">

**Reply by PhiloNL** [reply]
Thanks for the feedback! Then I'm probably mistaken 🤓

**Comment by jesperbjoernandersen** [comment]
I've been testing further, and it does seem like $dispatch sends the correct browser event with w/e payload you have. The issue seems to be picking up these events fired from $dispatch, in the frontend. I was even unable listen for the event using Alphine

**Comment by Aridez** [comment]
Hey there! 

I ran into this same problem when I upgraded to V3 and it was because I realized that the custom input needed to use `wire:model.live` in my case. My custom input component is now working with this code:

```html
    <x-body.steps wire:model.live="currentStep">
        <x-body.steps.item x-data @click="$dispatch('input', '1')">
            {{__("Step 1")}}
        </x-body.steps.item>
        <x-body.steps.item x-data @click="$dispatch('input', '2')">
            {{__("Step 2")}}
        </x-body.steps.item>
                <x-body.steps.item x-data @click="$dispatch('input', '3')">
            {{__("Step 3")}}
        </x-body.steps.item>
    </x-body.steps>
 ```

Hope this helps!
